src/app/clusters/energy-evse-server/energy-evse-server.h

```diff
@@ -38,13 +39,31 @@ namespace EnergyEvse {
 constexpr int64_t kMinimumChargeCurrent         = 0;
 constexpr int64_t kMaximumChargeCurrent         = 80000;
 constexpr uint32_t kMaxRandomizationDelayWindow = 86400;
+constexpr uint8_t kEvseTargetsMaxNumberOfDays   = 7;
+constexpr uint8_t kEvseTargetsMaxTargetsPerDay  = 10;
+
+struct EvseChargingTarget
+{
+    uint16_t targetTimeMinutesPastMidnight;
+    Optional<chip::Percent> targetSoC;
+    Optional<int64_t> addedEnergy;
+};
+
+class EvseTargetEntry
+{
+public:
+    chip::BitMask<TargetDayOfWeekBitmap> dayOfWeekMap;
+    std::vector<EvseChargingTarget> dailyChargingTargets;
```

Comment:
change to a span instead, do memory management in the next handler with a span to the size needed.

Reply:
No longer done with vectors. Closing this.